airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -152,15 +154,14 @@ def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapp
             if record.get(self.created_at_field) and self.str2datetime(record[self.created_at_field]) < self._start_date:
                 continue
             yield record
-        yield from []
 
     def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
         # try to save maximum value of a cursor field
-        return {
-            self.cursor_field: max(
-                str((latest_record or {}).get(self.cursor_field, "")), str((current_stream_state or {}).get(self.cursor_field, ""))
-            )
-        }
+        old_value = str((current_stream_state or {}).get(self.cursor_field, ""))
+        new_value = max(str((latest_record or {}).get(self.cursor_field, "")), old_value)
+        if old_value != new_value:
+            return {self.cursor_field: new_value}
+        return None
```

Comment:
why do we need to return None instead of current stream state ?

Reply:
not need. it was testing some assumption only. returned the standard behaviour